### PR TITLE
Modded payload factory sprites

### PR DESCRIPTION
**Pull request by MEEPofFaith**
Allow mods to have their own `factory-in/out/top-<size>(-suffix)` sprites.
Also made payload mass drivers have their outlines generated so that mods do not need to manually outline them.

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [X] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [X] I have ensured that my code compiles, if applicable.
- [X] I have ensured that any new features in this PR function correctly in-game, if applicable.
  - [X] Vanilla textures do not break (Ran `tools:pack`)
  - [X] Modded factory sprites work (Test with custom builds of [poly-1810/tantros-but-java](https://github.com/poly-1810/tantros-but-java))
    - [X] Renamed `payload-launcher-in/out` to `factory-in/out-2` and removed suffix from the payload launcher block
    - [X] Renamed `payload-launcher-in/out` to `factory-in/out-2-deep` and set the payload launcher block's suffix to `"-deep"`
  - [X] Modded payload mass driver sprites properly generate outlines

### CODE REVIEW

File: core/src/mindustry/world/blocks/payloads/PayloadBlock.java [hunk 1]
```diff
@@ -32,9 +32,18 @@ public PayloadBlock(String name){
...
 
-        topRegion = Core.atlas.find(name + "-top", "factory-top-" + size + regionSuffix);
-        outRegion = Core.atlas.find(name + "-out", "factory-out-" + size + regionSuffix);
-        inRegion = Core.atlas.find(name + "-in", "factory-in-" + size + regionSuffix);
+        topRegion = findFactoryRegion("-top");
+        outRegion =  findFactoryRegion("-out");
+        inRegion =  findFactoryRegion("-in");
+    }
+
+    protected TextureRegion findFactoryRegion(String suf){
```
Comment (MEEPofFaith) [inline core/src/mindustry/world/blocks/payloads/PayloadBlock.java:40]: Modeled after how `DrawTurret` finds the base region.

**Review by Yruma-kun — APPROVED** [review]

**Review by BlueTheCube — APPROVED** [review]
this is epic

File: core/src/mindustry/world/blocks/payloads/PayloadMassDriver.java [hunk 2]
```diff
@@ -122,8 +122,8 @@ public void drawPlace(int x, int y, int rotation, boolean valid){
     }
 
     @Override
-    public TextureRegion[] makeIconRegions(){
-        return new TextureRegion[]{leftRegion, rightRegion, capRegion};
+    public void getRegionsToOutline(Seq<TextureRegion> out){
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/payloads/PayloadMassDriver.java:125]: Why was this changed - was it tested to make sure things don't break?

Comment (MEEPofFaith) [reply core/src/mindustry/world/blocks/payloads/PayloadMassDriver.java:125]: I legitimately do not remember

Comment (BlueTheCube) [reply core/src/mindustry/world/blocks/payloads/PayloadMassDriver.java:125]: can you revert it

Comment (MEEPofFaith) [reply core/src/mindustry/world/blocks/payloads/PayloadMassDriver.java]: > Why was this changed - was it tested to make sure things don't break?

Oh, now I remember, it's literally in the commit name.
Since `makeIconRegions` is vanilla-only, modded ones wouldn't get outlined.
...I'm gonna go test real quick.

Comment (MEEPofFaith) [reply core/src/mindustry/world/blocks/payloads/PayloadMassDriver.java:126]: Yep. Reverted my reversion.

Before change:
20240624-214407-javaw

After change:
20240624-214506-javaw
